tpgtools/property.go

```diff
@@ -550,8 +568,12 @@ func createPropertiesFromSchema(schema *openapi.Schema, typeFetcher *TypeFetcher
 
 		if v, ok := v.Extension["x-dcl-conflicts"].([]interface{}); ok {
 			for _, ci := range v {
-				p.ConflictsWith = append(p.ConflictsWith, ci.(string))
+				cf := ci.(string)
+				cf = cf[strings.LastIndex(cf, ".")+1:]
+				p.JSONCaseConflictsWith = append(p.JSONCaseConflictsWith, cf)
 			}
+
+			conflictsMap[p.PackageJSONName()] = p.objectIdxPath()
```

Comment:
If we build the map like this, will conflicts work for nested objects? Theoretically we could have an object that has a field `foo` and a nested object with a field of the same name like `bar.foo`. This would cause unexpected conflicts here, right?

Reply:
Great catch! Currently, the DCL only generate `x-dcl-conflicts` on top-level fields, but not nested-fields (b/204789576). I made this fix under the assumption that conflict fields are either top-level fields or are on the same level with same parent field. Agree that it may not be the situation for all resources.

Reply:
Or should we just get a fix for top-level field for now and add TODO for nested conflict field until the DCL got the information generated. Any suggestions?

Reply:
Yeah, I would add it for top-level and throw an error if we find anything that isn't top-level so we know when we have to handle this

Reply:
Sure! will do that. Thank you so much!